xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/internal/authentication/LoginLinkUIExtension.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.internal.authentication;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.template.TemplateManager;
+import org.xwiki.uiextension.UIExtension;
+
+/**
+ * Injects the login link in the top bar.
+ *
+ * @version $Id$
+ */
+@Component(LoginLinkUIExtension.ROLE_HINT)
+@Singleton
+public class LoginLinkUIExtension implements UIExtension
+{
+    /**
+     * The role hint.
+     */
+    public static final String ROLE_HINT = "org.xwiki.platform.oldcore.loginlink";
```

Comment:
Yes, now the only point of this id is to be unique (a bit like a listener name), and changing it should not break anything.

Reply:
@surli I tried to follow https://dev.xwiki.org/xwiki/bin/view/Community/DevelopmentPractices#HUserInterfaceExtensionPointNaming which says we need to put the module name after the group id, and the module that currently provides these 2 UI extensions is ``oldcore``. If we move them we can rename the id, but I wouldn't use a different module name just because **we might** move these in the future.